### PR TITLE
allow rebase by image digest

### PR DESCRIPTION
**Pull request by joeybrown-sf**

### CODE REVIEW

File: text/0000-rebase-immutable-image-ref.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+# Motivation
+[motivation]: #motivation
+
+- This feature will support a workflow where the rebase executing agent only has the digest available to it.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This provides a high level overview of the feature.
```
Comment (jabrown85) [inline text/0000-rebase-immutable-image-ref.md]: I think you should fill this out

File: text/0000-rebase-immutable-image-ref.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing users and new users.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+First, let's get some context. Today, we can execute rebase by using **tag references** but not **digest references**.
+
+Here are some examples of valid rebase commands using **tag references**:
+1. `lifecycle rebase my-repo/foo`
```
Comment (jabrown85) [inline text/0000-rebase-immutable-image-ref.md:50]: is it `rebase` or `rebaser`? I don't recall. I do know the binary location is `/cnb/lifecycle/rebaser`

File: text/0000-rebase-immutable-image-ref.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+Here are some examples of currently invalid rebase commands using **digest references**:
+1. `lifecycle rebase my-repo/foo@sha256:1234 myrepo/foo`
+1. `lifecycle rebase my-repo/foo@sha256:1234 myrepo/foo:latest`
+1. `lifecycle rebase my-repo/foo@sha256:1234 myrepo/foo:vNext`
+
+Supporting Rebase by Image Digest Reference will make both sets of commands valid.
+
+- When using a digest reference as the image target, there must be at least one tag reference to apply to exported image.
+- **If** the `rebase target` is a `digest reference` and one or more `<tag reference>` inputs are provided, each `<tag reference>` MUST refer to the same repository as the `rebase target`.
```
Comment (jabrown85) [inline text/0000-rebase-immutable-image-ref.md]:
```suggestion
- **If** the `rebase target` is a `digest reference` and one or more `<tag reference>` inputs are provided, each `<tag reference>` MUST refer to the same repository as the `rebase target`.
```

I don't see why we want/need this. The additional tags can, today, be on any registry so long as the authentication mechanisms are in place to allow `rebaser` to push there.

File: text/0000-rebase-immutable-image-ref.md (hunk 2 above)
Comment (joeybrown-sf) [reply text/0000-rebase-immutable-image-ref.md:50]: it is rebase

File: text/0000-rebase-immutable-image-ref.md (hunk 1 above)
Comment (joeybrown-sf) [reply text/0000-rebase-immutable-image-ref.md]: thanks missed it!

**Pull request closed without merging**